worlds/osrs/__init__.py

```diff
@@ -0,0 +1,616 @@
+
+        woodcutting_tasks = [task for task in self.locations_by_category["Woodcutting"]
+                             if task.skills[0].level <= int(self.multiworld.max_woodcutting_level[self.player])]
+        if not self.multiworld.progressive_tasks[self.player]:
+            rnd.shuffle(woodcutting_tasks)
+        else:
+            woodcutting_tasks.reverse()
+        woodcutting_tasks = woodcutting_tasks[0:self.multiworld.max_woodcutting_tasks[self.player]]
+        woodcutting_weight = self.multiworld.woodcutting_task_weight[self.player] if len(woodcutting_tasks) > 0 else 0
+
+        all_tasks = [
+            combat_tasks,
+            prayer_tasks,
+            magic_tasks,
+            runecraft_tasks,
+            crafting_tasks,
+            mining_tasks,
+            smithing_tasks,
+            fishing_tasks,
+            cooking_tasks,
+            firemaking_tasks,
+            woodcutting_tasks,
+            general_tasks
+        ]
+        all_weights = [
+            combat_weight,
+            prayer_weight,
+            magic_weight,
+            runecraft_weight,
+            crafting_weight,
+            mining_weight,
+            smithing_weight,
+            fishing_weight,
+            cooking_weight,
+            firemaking_weight,
+            woodcutting_weight,
+            general_weight
+        ]
+
+        while locations_added < locations_required:
+            taskChosen = False
+            if all_tasks:
+                chosen_task = rnd.choices(all_tasks, all_weights)[0]
+                if len(chosen_task) > 0:
+                    task = chosen_task.pop()
+                    taskChosen = True
+                if len(chosen_task) == 0:
+                    index = all_tasks.index(chosen_task)
+                    del all_tasks[index]
+                    del all_weights[index]
+
+            else:
+                assert len(general_tasks) > 0, "There are not enough avaialbe tasks to vfill the remaining pool for OSRS"
+                task = general_tasks.pop()
+                pass
+            if taskChosen:
+                self.add_location(task)
+                locations_added += 1
+
+    def add_location(self, location):
+        index = [i for i in range(0, len(self.location_rows)) if self.location_rows[i].name == location.name][0]
+        self.create_and_add_location(index)
+
+    def create_items(self) -> None:
+        for item_row in self.item_rows:
+            if item_row.name != self.starting_area_item:
+                for c in range(item_row.count):
+                    item = self.create_item(item_row.name)
+                    self.multiworld.itempool.append(item)
+
+    def create_and_add_location(self, row_index) -> None:
+        location_row = self.location_rows[row_index]
+        # print(f"Adding task {location_row.name}")
+
+        # Create Location
+        id = self.base_id + row_index
```

Comment:
`id` is a standard python keyword, I recommend not using it as a variable name. Use `location_id` or something instead :)

(PyCharm for example complains here with the message "Shadows built-in name 'id'")